src/core/lib/address_utils/parse_address.cc

```diff
@@ -144,6 +147,60 @@ grpc_error_handle UnixAbstractSockaddrPopulate(
 }  // namespace grpc_core
 #endif  // GRPC_HAVE_UNIX_SOCKET
 
+#ifdef GRPC_HAVE_VSOCK
+
+bool grpc_parse_vsock(const grpc_core::URI& uri,
+                      grpc_resolved_address* resolved_addr) {
+  if (uri.scheme() != "vsock") {
+    gpr_log(GPR_ERROR, "Expected 'vsock' scheme, got '%s'",
+            uri.scheme().c_str());
+    return false;
+  }
+  grpc_error_handle error =
+      grpc_core::VSockaddrPopulate(uri.path(), resolved_addr);
+  if (!error.ok()) {
+    gpr_log(GPR_ERROR, "%s", grpc_core::StatusToString(error).c_str());
+    return false;
+  }
+  return true;
+}
+
+namespace grpc_core {
+
+grpc_error_handle VSockaddrPopulate(absl::string_view path,
```

Comment:
Although `UnixAbstractSockaddrPopulate` appears to use `grpc_error_handle`, we can use `absl::Status` instead here as we're recommended to use it if possible. `absl::Status` is still having debug-location for debug build so simply using `absl::Status` would be recommended. If capturing file & line number is a must, we can do it manually by adding `GRPC_DEFAULT_FILE` and `GRPC_DEFAULT_LINE` to the message but I guess it isn't.

Reply:
I thought about this, but since the other code in this file still uses the old status APIs, it seems fine for this code to do the same thing.  At some point we'll want to modernize all of the code in this file, but until then we might as well be consistent.

I would love it if someone wants to dive in and modernize this code, including moving all of it to idiomatic C++ APIs instead of old style C APIs.  But I didn't want to hold up this feature on that work.

Reply:
I have no strong opinion here, fine to leave as is. Thank you for the guidance regardless, Esun, that will be helpful going forward.